src/main/kotlin/org/koil/image/Storage.kt

```diff
@@ -0,0 +1,53 @@
+package org.koil.image
+
+import com.amazonaws.HttpMethod
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.AmazonS3Exception
+import com.amazonaws.services.s3.model.GeneratePresignedUrlRequest
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.amazonaws.services.s3.model.PutObjectRequest
+import java.io.InputStream
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.*
+
+interface Storage {
+    fun saveObject(id: UUID, file: InputStream, contentType: String)
+    fun getPresignedObjectUrl(id: UUID): String
+}
+
+class S3Storage(
+    private val s3Client: AmazonS3,
+    private val bucketName: String
+) : Storage {
+    init {
+        createBucketIfNotExists()
+    }
+
+    override fun saveObject(id: UUID, file: InputStream, contentType: String) {
```

Comment:
not a block comment- but should contentType be an enum? What values can it have?

Reply:
Is there any nice failure modes for if the image fails to save to s3? I would kind of expect this to have a ObjectSaveResponse return type with some details in case of failure- but maybe it's handled elsewhere that I haven't seen yet.